packages/ocular-ui/components/chat/chat-helpers/index.tsx

```diff
@@ -60,18 +49,59 @@ export const fetchChatResponse = async (
   setIsGenerating: React.Dispatch<React.SetStateAction<boolean>>,
   setChatMessages: React.Dispatch<React.SetStateAction<ChatMessage[]>>
 ) => {
-  const response = await api.chats.sendMessage(chat.id,{message:content,stream:false},cancelTokenSource)
-  if (!response.data.ok) {
-    if (response.status === 404) {
-      toast.error(
-        "Failed To Chat",
-      )
-    }
-    const errorData = await response.data
-    toast.error(errorData.message)
-    setIsGenerating(false)
-  }
-  return response.data
+
+  let idx=0;
+  api.chats.sendMessage(chat.id,{message:content,stream:true},cancelTokenSource)
+        .then(async response => {
+          console.log("Streaming Copilot Response")
+          const reader = createReader(response.body);
+          const chunks = readStream(reader);
+          for await (const chunk of chunks) {
+            setChatMessages(prevChatMessages => {
+              const updatedMessages = [...prevChatMessages];
+              if (idx=== 0) {
+                updatedMessages.push({
+                  message: {
+                    chat_id: chunk.metadata?.chat_id,
+                    content: chunk.choices[0].delta.content,
+                    created_at: new Date(),
+                    id: "123",
+                    role: chunk.metadata.role,
+                    updated_at: new Date(),
+                    user_id: chunk.metadata.user_id
+                  },
+                  fileItems: []
+                });
+                idx++;
+              } else {
+                const lastMessage = updatedMessages[updatedMessages.length - 1];
+                lastMessage.message.content = chunk.choices[0].delta.content;
+                lastMessage.message.updated_at = new Date();
+              }
+              return updatedMessages;
+            });
+          }
+          
+          // setChatMessages(prevChatMessages => [...prevChatMessages, 
```

Comment:
@Vivek-Lahole do we still need this too?

Reply:
no dont need that